src/StakeRegistry.sol

```diff
@@ -347,6 +359,8 @@ contract StakeRegistry is StakeRegistryStorage {
                     nextUpdateBlockNumber: 0,
                     stake: newStake
                 }));
+
+                _updateQuorumRoot(operatorId, quorumNumber, uint32(block.number), newStake);
```

Comment:
Should be called below, before the function returns. Currently, this misses a branch where the operator's stake is updated twice in one block.